contractcourt/chain_watcher_test.go

```diff
@@ -213,3 +223,154 @@ func TestChainWatcherRemoteUnilateralClosePendingCommit(t *testing.T) {
+		// call into this channel so we can feed in arbitrary state
+		// deviations at all, without doing any state updates.
+		stateNumChan := make(chan uint64, 1)
+
+		// With the channels created, we'll now create a chain watcher
+		// instance which will be watching for any closes of Alice's
+		// channel.
+		aliceNotifier := &mockNotifier{
+			spendChan: make(chan *chainntnfs.SpendDetail),
+		}
+		aliceChainWatcher, err := newChainWatcher(chainWatcherConfig{
+			chanState: aliceChannel.State(),
+			notifier:  aliceNotifier,
+			signer:    aliceChannel.Signer,
+			extractStateNumHint: func(*wire.MsgTx,
+				[lnwallet.StateHintSize]byte) uint64 {
+
+				return <-stateNumChan
+			},
+		})
+		if err != nil {
+			t.Fatalf("unable to create chain watcher: %v", err)
+		}
+		if err := aliceChainWatcher.Start(); err != nil {
+			t.Fatalf("unable to start chain watcher: %v", err)
+		}
+		defer aliceChainWatcher.Stop()
+
+		// Based on the number of random updates for this state, make a
+		// new HTLC to add to the commitment, and then lock in a state
+		// transition.
+		const htlcAmt = 1000
+		for i := 0; i < int(numUpdates); i++ {
+			addFakeHTLC(
+				t, 1000, uint64(i), aliceChannel, bobChannel,
+			)
+
+			err := lnwallet.ForceStateTransition(
+				aliceChannel, bobChannel,
+			)
+			if err != nil {
+				t.Errorf("unable to trigger state "+
+					"transition: %v", err)
+				return false
+			}
+		}
+
+		// We'll request a new channel event subscription from Alice's
+		// chain watcher so we can be notified of our fake close below.
+		chanEvents := aliceChainWatcher.SubscribeChannelEvents()
+
+		// Now that we have our updates in place, we'll feed Alice a
+		// random state number. If this is beyond her best known state
+		// for Bob, then she should enter the DLP protocol.
+		stateNumChan <- uint64(broadcastStateNum)
+
+		// Otherwise, we'll feed in this new state number as a response
+		// to the query, and insert the expected DLP commit point.
+		dlpPoint := aliceChannel.State().IdentityPub
+		err = aliceChannel.State().MarkDataLoss(dlpPoint)
+		if err != nil {
+			t.Errorf("unable to insert dlp point: %v", err)
+			return false
+		}
+
+		// Now we'll trigger the channel close even to trigger the
+		// scenario.
+		bobCommit := bobChannel.State().LocalCommitment.CommitTx
+		bobTxHash := bobCommit.TxHash()
+		bobSpend := &chainntnfs.SpendDetail{
+			SpenderTxHash: &bobTxHash,
+			SpendingTx:    bobCommit,
+		}
+		aliceNotifier.spendChan <- bobSpend
+
+		// We should get a new uni close resolution that indicates we
+		// processed the DLP scenario.
+		var uniClose *lnwallet.UnilateralCloseSummary
+		select {
+		case uniClose = <-chanEvents.RemoteUnilateralClosure:
+			// If we processed this as a DLP case, then the remote
+			// party's commitment should be blank, as we don't have
+			// this up to date state.
+			blankCommit := channeldb.ChannelCommitment{}
+			if uniClose.RemoteCommit.FeePerKw != blankCommit.FeePerKw {
```

Comment:
also check that the commitPoint/contractResolution is properly populated to sweep our output

Reply:
Proper population of the entire uni close resolution is already tested within the `lnwallet` package. Added a clause here to ensure that it uses the fake commit point we passed in to derive their final sweeping keys.